src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -92,47 +83,56 @@ private function completeDefinition($id, Definition $definition)
         if (!$constructor = $reflectionClass->getConstructor()) {
             return;
         }
+        $parameters = $constructor->getParameters();
+        if (method_exists('ReflectionMethod', 'isVariadic') && $constructor->isVariadic()) {
+            array_pop($parameters);
+        }
 
         $arguments = $definition->getArguments();
-        foreach ($constructor->getParameters() as $index => $parameter) {
+        foreach ($parameters as $index => $parameter) {
             if (array_key_exists($index, $arguments) && '' !== $arguments[$index]) {
```

Comment:
I wonder if we actually need this, it makes one more thing to learn while we could achieve the same using manual indexes or named arguments.